src/js/components/AppListComponent.jsx

```diff
@@ -81,12 +114,42 @@ var AppListComponent = React.createClass({
     });
   },
 
+  getAppsInContext: function () {
+    var apps = this.state.apps;
+    var currentGroup = this.props.currentGroup;
+
+    return lazy(apps)
+      .filter((app) => app.id.startsWith(currentGroup))
+      .reduce((memo, app) => {
+        let relativePath = app.id.substring(currentGroup.length);
+        let pathParts = relativePath.split("/");
+        let isGroup = pathParts.length > 1;
+
+        if (!isGroup) {
+          memo.push(app);
+        } else {
+          let groupId = currentGroup + pathParts[0];
+          let group = memo.find((item) => {
+            return item.id === groupId;
+          });
+
+          if (group == null) {
+            group = createGroup(groupId, app);
+            memo.unshift(group);
+          } else {
+            updateGroup(group, app);
+          }
+        }
+        return memo;
+      }, []);
+  },
+
   getAppNodes: function () {
     var state = this.state;
     var sortKey = state.sortKey;
     var props = this.props;
 
-    var appsSequence = lazy(state.apps);
+    var appsSequence = lazy(this.getAppsInContext());
```

Comment:
If ```lazy().reduce()``` from ```getAppsInContext()``` returns a real value instead of a sequence, then please ignore this comment. :)
Otherwise, if reduce returns a sequence, this lazy() here isn't needed..
The documentation of lazy.js isn't too concise about this.